Modules/Filtering/ImageGradient/test/itkGradientMagnitudeImageFilterTest.cxx

```diff
@@ -38,6 +40,19 @@ itkGradientMagnitudeImageFilterTest(int, char *[])
     itk::GradientMagnitudeImageFilter<ImageType, ImageType>::Pointer filter =
       itk::GradientMagnitudeImageFilter<ImageType, ImageType>::New();
 
+    bool useImageSpacing = true;
```

Comment:
Did you mean to ahve something like "auto useImageSpacing = static_cast<bool>(std::stoi(argv[1]));"

else block at 49-52 is never used.

Reply:
In this case, contrary to [here](https://github.com/InsightSoftwareConsortium/ITK/pull/2286#discussion_r578092375), I grant that the previous commit did not add an input argument to the test, and I now realize that some compilers might flag the blocks as never being reached, which is annoying. 

So I added in 8e4315a a test with each boolean value taking advantage of the fact that there is no comparison to a baseline in this case. That can be added at a latter time and see if the results make sense.